eng/linkerTests/project.csproj.template

```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Does this setup mean that once there is a test for RID scenario it will be run with every RID we have in our build matrix?

Reply:
For now, we are only running the tests in windows-x64, linux-x64, and osx-x64. The plan is that as we start getting some tests that need to be covered in different RID/platform, then we will just add builds for those platforms.

Reply:
None of these really matter for .NET5 so there should perhaps be different prioritization (@danmosemsft)

Reply:
For these tests we don't really care much about the actual runtime that the end user will have for .NET 5 as much, what we really care about is that the attributes are placed correctly such that the linker will not strip methods that you would use during execution. For this, we want to to cover places where the libraries code might use one branch of code or the other, which for our runtime libraries most of those branches are dependent on whether you are running on Windows, Linux or OSX (hence the initial set of runtimes we are testing). Once we start adding code branches that would only ever be touched in wasm, then we would definitely care about adding that to our runtimes set as well as we for sure want to validate that the attributes are placed on the right places in there as well. I'm happy to chat more about this if you want, but the high level explanation is that these tests are not as much to test the end product experience or scenario, but more of tests for our annotations correctness.

Reply:
> Once we start adding code branches that would only ever be touched in wasm

It's not once but that's the setup today. We build a very different version of libraries for browser today than what is CoreCLR version on desktop (starting with System.Private.CoreLib to libraries like System.Net.Http)

Reply:
Right, for wasm in particular we are simply not adding that with this PR because that is not ready yet, there is an ongoing PR that is adding support for running tests on wasm and once that is in, then we will add that to these sets of tests.